cms/djangoapps/contentstore/views/item.py

```diff
@@ -786,6 +852,11 @@ def safe_get_username(user_id):
         visibility_state = None
     published = modulestore().has_published_version(xblock) if not is_library_block else None
 
+    #instead of adding a new feature directly into xblock-info, we should add them into override_type.
+    override_type = {}
+    if getattr(xblock, "is_entrance_exam", None):
```

Comment:
This isn't really what I was thinking. An 'override_type' would be a single string representing the true type of the object (i.e. 'entrance_exam') rather than what the xblock type is (i.e. 'chapter'). This seems like a more generic way to provide additional fields to xblock info which I think is too flexible.

I can let this go for the sake of getting the release out, but I'd like to see this done more cleanly.